### PR TITLE
Rust 1.83, allow -> expect (missing_docs)

### PR DESCRIPTION
**Pull request by bas-ie**
# Objective

We were waiting for 1.83 to address most of these, due to a bug with `missing_docs` and `expect`. Relates to, but does not entirely complete, #15059.

## Solution

- Upgrade to 1.83
- Switch `allow(missing_docs)` to `expect(missing_docs)`
- Remove a few now-unused `allow`s along the way, or convert to `expect`

### CODE REVIEW

**Comment by richchurcher** [comment]
Let's see what the CI has to say now :eyes:

File: crates/bevy_reflect/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,4 @@
-// FIXME(15321): solve CI failures, then replace with `#![expect()]`.
-#![allow(missing_docs, reason = "Not all docs are written yet, see #3492.")]
-// `rustdoc_internals` is needed for `#[doc(fake_variadics)]`
+#![expect(missing_docs, reason = "Not all docs are written yet, see #3492.")]
 #![allow(internal_features)]
```
Comment (BenjaminBrienen) [inline crates/bevy_reflect/src/lib.rs:4]: Why did you remove this? It might have warned about being unused because it is only used with that feature enabled. Maybe put this under a cfg_attr instead?

**Review by BenjaminBrienen — CHANGES_REQUESTED** [review]

**Comment by BenjaminBrienen** [comment]
I think you can use expect there if you use cfg_attr.

File: crates/bevy_reflect/src/lib.rs (hunk 1 above)
Comment (richchurcher) [reply crates/bevy_reflect/src/lib.rs:4]: I see! Is https://github.com/bevyengine/bevy/pull/16561/commits/229ef7f5317b8a44c8019798b8a5620ff128c27a what you were thinking?

File: crates/bevy_render/src/lib.rs [hunk 2]
```diff
@@ -1,8 +1,5 @@
-// FIXME(15321): solve CI failures, then replace with `#![expect()]`.
-#![allow(missing_docs, reason = "Not all docs are written yet, see #3492.")]
-#![allow(unsafe_code)]
-// `rustdoc_internals` is needed for `#[doc(fake_variadics)]`
-#![allow(internal_features)]
```
Comment (BenjaminBrienen) [inline crates/bevy_render/src/lib.rs:5]: Do the same thing here :)

**Review by BenjaminBrienen — CHANGES_REQUESTED** [review]

File: crates/bevy_render/src/lib.rs (hunk 2 above)
Comment (BenjaminBrienen) [inline crates/bevy_render/src/lib.rs:5]: Do the same thing here :)

```suggestion
#![cfg_attr(any(docsrs, docsrs_dep), expect(internal_features))]
```

**Review by BenjaminBrienen — CHANGES_REQUESTED** [review]

**Review by BenjaminBrienen — APPROVED** [review]

**Review by BenjaminBrienen — APPROVED** [review]

File: examples/2d/bloom_2d.rs [hunk 3]
```diff
@@ -65,6 +65,7 @@ fn setup(
             left: Val::Px(12.0),
             ..default()
         },
+        TextColor(Color::srgb(5., 5., 5.)),
```
[the remark this answers is not in the document]
Comment (bas-ie) [reply examples/2d/bloom_2d.rs:68]: :thinking: why _did_ I do that?

[the remark this answers is not in the document]
Comment (bas-ie) [reply examples/2d/bloom_2d.rs]: I have absolutely no idea, must've snuck in while I was playing around with something! Ty for the spot.

**Review by BenjaminBrienen — APPROVED** [review]
